sherpa/fit.py

```diff
@@ -449,17 +491,17 @@ def __init__(self, data, model, stat, method, itermethod_opts=None):
         # models into the objects needed for simultaneous fitting,
         # if they are not already in such objects.
         self.data = data
-        if (type(data) is not DataSimulFit):
+        if type(data) is not DataSimulFit:
```

Comment:
I decided that I didn't want to address these issues in this PR. I felt it was big enough as is!

There's a bunch of areas like this - as you note below - that should be cleaned up, but I am not convinced that we yet have a reliable enough set of tests within sherpa to validate any such changes.